src/containers/app.js

```diff
@@ -37,6 +37,12 @@ import SafeAreaContainer from './SafeAreaContainer'
 import { setPageShowing } from '../actions/navBar'
 import NavBar from '../components/NavBar';
 import PageKeys from '../constants/PageKeys'
+import * as Sentry from '@sentry/react-native';
+
+Sentry.init({
+    dsn: 'https://334e2b2ca1c04dc4a7fc356e394e9ea8@o274434.ingest.sentry.io/5371400',
+    enableNative: process.env.NODE_ENV === 'production' ? true : false,
+});
```

Comment:
This definitely works. It's pretty cool Sentry will start reporting with a quick `.init()` at the root. 

However, this is currently reporting non-crashing bugs and warnings. It might be worthwhile to add an option here to only report crashing bugs. More info in the PR review.

Reply:
So, this one is interesting. I researched several things, and here's what I found:

1. Sentry has a couple of options that you can include in the `init` to manage what is sent, but none of them is specifically a log level for messages from the app itself. It has:

`logLevel` - this is not actually for the level of logs _from our app_. It is for the level of logs that Sentry should show us if Sentry _itself_ fails to send our app's errors through to its console.

`ignoreErrors` - this allows us to ignore _specific_ errors. That said, it's for errors—not warnings. It also requires us to know, verbatim, what we wish to exclude.

`error_types` - Only exists according to the documentation. I did not find it upon a thorough investigation of the ReactNativeOptions interface and all the interfaces it inherits from (`BrowserOptions`, `Options`). Supposedly, inherits a bytemask system from PHP that allows you to specify exactly which kinds of errors you want (defaulting to `E_ALL`). Like I said, does not exist in the implementation. Only mentioning in case you decide to do your own check of the documentation and you find this.

2. There's a manual approach. We can initialize Sentry with a pre-post block called `beforeSend` as [demonstrated here](https://docs.sentry.io/error-reporting/configuration/filtering/). However:

- It doesn't actually _stop_ Sentry from sending the event. It just allows us to modify it—say, by adding a tag, which theoretically means that the event would then only appear once _in the UI that we see_. I don't know for sure that it affects how each event counts toward our monthly limit.

3. Finally...it actually appears that Sentry _doesn't_ log warnings. It only logs errors. The "no avatar" situation is, it turns out, a full-blown error. It says here `type:error`:

![error](https://chelseatroy.com/wp-content/uploads/2020/08/Screen-Shot-2020-08-10-at-4.08.01-PM.png)

Furthermore, when I run the app, I see all of the following warnings, and none of those appear in Sentry:

![warnings](https://chelseatroy.com/wp-content/uploads/2020/08/Screen-Shot-2020-08-10-at-4.04.22-PM.png)

So I think the solution here might be to handle this particular error. Do we have any data on what proportion of our users do not have avatar images?

Reply:
Nice! Looks like there's a lot of good stuff covered here to help us figure out a good solution. I'm not sure what proportion of our users have avatar images, but I think we should _assume_ many of them do not. Handling this particular error is probably the best bet here.

I'm thinking there are several other errors hidden throughout the app similar to the avatar error that would log to sentry. Ex: I found another one while reviewing this PR related to signing out. The best solution here may be to add Sentry to the app, and then play around with the app in staging to audit that app and find out what errors are submitted to Sentry. We can then handle those errors in a separate PR as it's out of scope of this PR.

Otherwise, I can see a situation where we breeze through out Sentry allotment quickly without gracefully handling errors.

Reply:
I agree—I went ahead and fixed that error (see below comment).

I also did a little manual testing: I went into the app and clicked all around. I looked at tutorials. I looked at field guides. I performed classifications. Then I logged out. I didn't see any more errors.

That said, I _do_ see the error you're mentioning in Sentry from the original review: "Error: Failed to sign out; not signed in." How, uh...where in the UI did you find a sign out button while not signed in? We probably shouldn't have that 😂

Reply:
Cool! Then if that's the only exception that needs to be handled I suppose it can be dealt with in this PR

Reply:
Rock on. I tried to reproduce signing out while signed out, but haven't so far. So I'll merge this in, and if we figure out how to repro that, I'll fix it in a new PR!